src/main/java/ui/components/NavigableListView.java

```diff
@@ -111,25 +114,43 @@ public void restoreSelection() {
                 selectedIndex = Optional.of(nextIndex);
 
                 // The next item will be considered selected
-                onItemSelected.accept(nextIndex);
+                setItemSelectedParamsForPrimaryKey(nextIndex);
             }
         }
     }
 
+    /**
+     * Passes the relevant index to the onItemSelected callback variable.
+     * This method is called for primary keys e.g. keyboard keys and the left mouse key
+     */
+    private void setItemSelectedParamsForPrimaryKey(int index){
```

Comment:
`triggerItemSelected` (reasonable default, hence shorter name) and `triggerItemSelectedSecondary`?